boards/nucleo-f746zg/include/periph_conf.h

```diff
@@ -44,9 +103,9 @@ static const uart_conf_t uart_config[] = {
         .tx_af      = GPIO_AF7,
         .bus        = APB1,
         .irqn       = USART3_IRQn,
-#ifdef UART_USE_DMA
-        .dma_stream = 6,
-        .dma_chan   = 4
+#ifdef MODULE_PERIPH_DMA
```

Comment:
`UART_USE_DMA` is still used by other boards, so I'd keep it.

Reply:
I think `UART_USE_DMA` is a leftover from previous refactoring and it should be `MODULE_PERIPH_DMA`, like it is for most of the other nucleo (and stm32 based boards):

```
$ git grep MODULE_PERIPH_DMA boards/{stm32*,nucleo*,p-*,b-*} | wc -l
72
$ git grep UART_USE_DMA boards/{stm32*,nucleo*,p-*,b-*} | wc -l
14
```

The DMA feature is disabled by default. It's only tested by `tests/periph_dma` and the only way to verify it automatically is to use MODULE_PERIPH_DMA. With UART_USE_DMA, one would have to use CFLAGS.

@hugueslarrive can you post the output of `make BOARD=nucleo-f746zg -C tests/periph_dma flash test` ?

Reply:
All other boards that features periph_dma have  `#ifdef MODULE_PERIPH_DMA` rather than `#ifdef UART_USE_DMA` in their `periph_conf.h`.

```
$ grep -lr UART_USE_DMA
boards/nucleo-l433rc/include/periph_conf.h
boards/nucleo-f303ze/include/periph_conf.h
boards/stm32f769i-disco/include/periph_conf.h
boards/nucleo-f746zg/include/periph_conf.h
boards/stm32l476g-disco/include/periph_conf.h
boards/nucleo-f722ze/include/periph_conf.h
boards/nucleo-l4r5zi/include/periph_conf.h
boards/nucleo-l496zg/include/periph_conf.h
```
There is nothing in the cpu directory to handle UART_USE_DMA rather than MODULE_PERIPH_DMA:
```
$ grep -lr MODULE_PERIPH_DMA
[...]
cpu/stm32/include/periph_cpu.h
cpu/stm32/cpu_init.c
cpu/stm32/periph/uart.c
cpu/stm32/periph/spi.c
cpu/saml21/cpu.c
cpu/saml1x/cpu.c
cpu/sam0_common/include/periph_cpu_common.h
cpu/sam0_common/periph/spi.c
cpu/samd5x/cpu.c
cpu/samd21/cpu.c
```